vnext/Microsoft.ReactNative/Views/FlyoutViewManager.cpp

```diff
@@ -358,6 +358,39 @@ winrt::Flyout FlyoutShadowNode::GetFlyout() {
   return m_flyout;
 }
 
+void FlyoutShadowNode::OnShowFlyout() {
+  AdjustDefaultFlyoutStyle(50000, 50000);
+  if (m_isFlyoutShowOptionsSupported) {
+    m_flyout.ShowAt(m_targetElement, m_showOptions);
+  } else {
+    winrt::FlyoutBase::ShowAttachedFlyout(m_targetElement);
+  }
+
+  auto popup = GetFlyoutParentPopup();
+  if (popup != nullptr) {
+    popup.IsLightDismissEnabled(m_isLightDismissEnabled);
+  }
+
+  if (m_isLightDismissEnabled) {
+    EnsureXamlRootChangedRevoker();
+  }
+}
+
+void FlyoutShadowNode::EnsureXamlRootChangedRevoker() {
+  if (m_xamlRootChangedRevoker) {
+    return;
+  }
+  if (auto flyoutBase6 = m_flyout.try_as<winrt::IFlyoutBase6>()) {
+    auto wkinstance = GetViewManager()->GetReactInstance();
+    if (auto instance = wkinstance.lock()) {
+      if (auto xamlRoot = static_cast<NativeUIManager *>(instance->NativeUIManager())->tryGetXamlRoot()) {
+        m_xamlRootChangedRevoker =
+            xamlRoot.Changed(winrt::auto_revoke, [this](auto &&, auto &&) { onDropViewInstance(); });
```

Comment:
Assumption here seems to be that IsLightDismissEnabled is true. Which you handle by hooking/unhooking the event. You also could just _check_
```
if (IsLightDismissEnabled) {
    onDropViewInstance();
}
```

Reply:
@chrisglein thanks, I like it! it will simplify things.